app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -48,34 +53,103 @@ import org.mozilla.fenix.wallpapers.Wallpaper
     ) {
-        WallpaperThumbnails(
-            wallpapers = wallpapers,
-            defaultWallpaper = defaultWallpaper,
-            loadWallpaperResource = loadWallpaperResource,
-            selectedWallpaper = selectedWallpaper,
-            onSelectWallpaper = { updatedWallpaper -> onSelectWallpaper(updatedWallpaper) },
-        )
+        wallpaperGroups.forEach { (collection, wallpapers) ->
+            if (wallpapers.isNotEmpty()) {
+                WallpaperGroupHeading(
+                    collection,
+                    onLearnMoreClick,
+                )
+                WallpaperThumbnails(
+                    wallpapers = wallpapers,
+                    defaultWallpaper = defaultWallpaper,
+                    loadWallpaperResource = loadWallpaperResource,
+                    selectedWallpaper = selectedWallpaper,
+                    onSelectWallpaper = onSelectWallpaper,
+                    verticalPadding = 0,
+                    horizontalPadding = 18,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun WallpaperGroupHeading(
+    collection: Wallpaper.Collection,
+    onLearnMoreClick: (String) -> Unit,
+) {
+    Column(
+        modifier = Modifier.padding(
+            top = 16.dp,
+            start = 18.dp,
+            end = 18.dp,
+            bottom = 12.dp,
+        ),
+    ) {
+        // Since the last new collection of wallpapers was tied directly to an MR release,
+        // it was decided that we should use string resources for these titles
+        // and descriptions so they could be localized.
+        // In the future, we may want to either use the dynamic wallpaper properties with localized fallbacks
+        // or invest in a method of localizing the remote strings themselves.
+        if (collection.name != "classic-firefox") {
+            Text(
+                text = stringResource(R.string.seasonal_wallpaper_title),
+                modifier = Modifier.padding(bottom = 2.dp),
+                color = FirefoxTheme.colors.textSecondary,
+                style = FirefoxTheme.typography.subtitle2,
+            )
+
+            Row(modifier = Modifier.fillMaxWidth()) {
+                Text(
+                    text = stringResource(R.string.seasonal_wallpaper_description),
+                    textAlign = TextAlign.Start,
+                    color = FirefoxTheme.colors.textSecondary,
+                    style = FirefoxTheme.typography.caption,
+                )
+
+                collection.learnMoreUrl?.let {
+                    Text(
+                        text = stringResource(R.string.wallpaper_learn_more),
+                        color = FirefoxTheme.colors.textSecondary,
+                        modifier = Modifier.clickable { onLearnMoreClick(collection.learnMoreUrl) },
+                        style = FirefoxTheme.typography.caption.copy(
+                            textDecoration = TextDecoration.Underline,
```

Comment:
Let's add the `textDecoration` outside the TextStyle since `Text` has a `textDecoration` parameter anyways